src/server/conn.rs

```diff
@@ -681,7 +681,7 @@ where
 
     /// Prevent shutdown of the underlying IO object at the end of service the request,
     /// instead run `into_parts`. This is a convenience wrapper over `poll_without_shutdown`.
-    pub fn without_shutdown(self) -> impl Future<Output = crate::Result<Parts<I, S>>>
+    pub async fn without_shutdown(self) -> crate::Result<Parts<I, S>>
```

Comment:
I think technically this a breaking change, since an async fn is automatically `!Unpin`, right?

Reply:
Forgot that the auto trait propagation made the earlier future `Unpin`, despite it not being explicitly set. Reverted